resources/tekton-config.json

```diff
@@ -0,0 +1,29 @@
+{
+    "spec": {
+        "pipeline": {
+            "enable-bundles-resolver": true,
+            "enable-cluster-resolver": true,
+            "enable-custom-tasks": true,
+            "enable-git-resolver": true,
+            "enable-hub-resolver": true,
+            "enable-tekton-oci-bundles": true
+        },
+        "chain": {
+            "artifacts.oci.storage": "oci",
+            "artifacts.pipelinerun.format": "in-toto",
+            "artifacts.pipelinerun.storage": "oci",
+            "artifacts.taskrun.format": "in-toto",
+            "artifacts.taskrun.storage": "oci",
+            "transparency.enabled": "true"
+        },
+        "platforms": {
+            "openshift": {
+                "pipelinesAsCode": {
+                    "settings": {
+                        "remember-ok-to-test": "false"
```

Comment:
fwiw while RHTAP went with this, in part because of some back and forth wrt Konflux, this feature was not yet enabled in Konflux (I'll avoid the tl;dr and explain in the next parking lot if you like), as the Pipelines-as-code SME in openshift-pipelines / tekton did not necessarily agree with the approach

All that said, I'm not suggesting breaking away from what RHTAP is doing.

But there is some interesting devil in the details wrt this.

Lastly, cross reference with my other comment, the current Konflux Tekton Config:  https://github.com/redhat-appstudio/infra-deployments/blob/c06c88f16fbd5998ad9465ae0cc2612f0ffee248/components/pipeline-service/development/main-pipeline-service-configuration.yaml#L1631-L1840

Reply:
> > fwiw while RHTAP went with this, in part because of some back and forth wrt Konflux, this feature was not yet enabled in Konflux (I'll avoid the tl;dr and explain in the next parking lot if you like), as the Pipelines-as-code SME in openshift-pipelines / tekton did not necessarily agree with the approach
> > All that said, I'm not suggesting breaking away from what RHTAP is doing.
> > But there is some interesting devil in the details wrt this.
> > Lastly, cross reference with my other comment, the current Konflux Tekton Config: https://github.com/redhat-appstudio/infra-deployments/blob/c06c88f16fbd5998ad9465ae0cc2612f0ffee248/components/pipeline-service/development/main-pipeline-service-configuration.yaml#L1631-L1840
> 
> @gabemontero I agree that I think its worth updating our tekton config to match the most recent one, though I wonder if this should be a separate task to avoid set backs in compatibility with our templates, wanna make sure it works as it did with the rhtap-installer. Though I am going to try to remove `openshift.pipelinesAsCode.settings.remember-ok-to-test`, as I suspect it won't matter from my understanding of what this does.

Tested removing `openshift.pipelinesAsCode.settings.remember-ok-to-test` and found behaviour is the same, opted to remove this to keep consistency: https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/7/commits/fa8bec73487eb86784456959ff288f94ff59a927

Reply:
Interesting @michael-valdron .... perhaps an openshift pipelines bug :-).

A minor follow up suggestion: I'm fine if you do not want to pursue, but if
- you converted this json file to its yaml alternative, so you can then put comments (https://stackoverflow.com/questions/244777/can-comments-be-used-in-json)
- and then you could put a comment explaining your findings here, so we have some historical reference "in the code" that explains things if a question in this space comes up in the future


WDYT?

thanks

Reply:
@gabemontero  Unfortunately, `kubectl` does not recognize the `_comment` property as a comment property to ignore:
```
Error from server (BadRequest): admission webhook "webhook.operator.tekton.dev" denied the request: mutation failed: cannot decode incoming new object: json: unknown field "_comment"
FAIL
```

To provide comments I've decided to convert this into a yaml file that gets processed into json with `yq` (already does this) to be passed into the `kubectl patch` command to patch the TektonConfig CR: https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/7/commits/f626d13969115ab60298755cd7fb1f804ec78ae3

With this, I've provided the comment you requested as a reference point in case we want to revert for any reason.